src/base/DeadCodeDetector.ml

```diff
@@ -0,0 +1,428 @@
+                (* m is a field, thus we don't track its liveness *)
+                (live_vars, adts)
+              else (
+                warn "Unused load statement to: " x ER.get_loc;
+                (live_vars, adts))
+          | RemoteLoad (x, addr, m) ->
+              mark_field_read m;
+              (* m is a field, thus we don't track its liveness *)
+              if SCIdentifier.is_mem_id x live_vars then
+                (dedup_id_list (addr :: live_vars), adts)
+              else (
+                warn "Unused remote load statement to: " x ER.get_loc;
+                (live_vars, adts))
+          | Store (i, m) ->
+              mark_field_write i;
+              (dedup_id_list (m :: live_vars), adts)
+          | MapUpdate (i, il, io) ->
+              mark_field_write i;
+              let live_vars' =
+                match io with Some ii -> i :: ii :: il | None -> i :: il
+              in
+              (dedup_id_list @@ live_vars' @ live_vars, adts)
+          | MapGet (x, i, il, _) ->
+              mark_field_read i;
+              if SCIdentifier.is_mem_id x live_vars then
+                (dedup_id_list (i :: (il @ live_vars)), adts)
+              else (
+                warn "Unused map get statement to: " x ER.get_loc;
+                (live_vars, adts))
+          | RemoteMapGet (x, addr, i, il, _) ->
+              mark_field_read i;
+              if SCIdentifier.is_mem_id x live_vars then
+                (dedup_id_list (addr :: i :: (il @ live_vars)), adts)
+              else (
+                warn "Unused remote map get statement to: " x ER.get_loc;
+                (live_vars, adts))
+          | ReadFromBC (x, _) ->
+              if not @@ SCIdentifier.is_mem_id x live_vars then
+                warn "Unused Read From BC statement to: " x ER.get_loc;
+              (live_vars, adts)
+          | Throw topt -> (
+              match topt with
+              | Some x -> (dedup_id_list (x :: live_vars), adts)
+              | None -> (live_vars, adts))
+          | CallProc (p, al) ->
+              proc_dict := p :: !proc_dict;
+              (dedup_id_list (al @ live_vars), adts)
+          | Iterate (l, p) ->
+              proc_dict := p :: !proc_dict;
+              (dedup_id_list (l :: live_vars), adts)
+          | Bind (i, e) ->
+              let live_vars_no_i =
+                List.filter
+                  ~f:(fun x -> not @@ SCIdentifier.equal i x)
+                  live_vars
+              in
+              if SCIdentifier.is_mem_id i live_vars then
+                let e_live_vars, adts' = expr_iter e in
+                ( dedup_id_list @@ e_live_vars @ live_vars_no_i,
+                  dedup_name_list @@ adts' @ adts )
+              else (
+                warn "Unused bind statement to: " i ER.get_loc;
+                (live_vars, adts))
+          | MatchStmt (i, pslist) ->
+              let live_vars', adts' =
+                List.fold_left pslist ~init:(live_vars, adts)
```

Comment:
This doesn't seem right. There shouldn't be any live variables when you start analysing a `match` branch. (You've got it correct for `MatchExpr`). Maybe a test case for this too?